wrapanapi/systems/msazure.py

```diff
@@ -653,21 +653,39 @@ def list_resource_groups(self):
         """
         return [r.name for r in self.resource_client.resource_groups.list()]
 
-    def list_free_pip(self, pip_template, resource_group=None):
+    def list_free_pip(self, pip_template=None, resource_group=None):
         """
-        List available Public IP in selected resource_group
+        List all Public IPs or specific VMs available Public IP in selected resource_group
         """
         resource_group = resource_group or self.resource_group
         ips = self.network_client.public_ip_addresses.list(resource_group_name=resource_group)
-        return [ip.name for ip in ips if not ip.ip_configuration and pip_template in ip.name]
+        all_free_pips = [ip.name for ip in ips if not ip.ip_address]
```

Comment:
There is a significant change here from `ip.ip_configuration` to `ip.ip_address` to determine whether an address is free.

Right now, in the SATQE resource group, nothing is returned for this, but there are several Public IP resources that are not attached to anything, which is what I think this should still return.

With your change there are no PIP's returned, with all your changes but this looking at `ip.ip_configuration` 8 PIPs are returned.

Reply:
Ohh really !

It returns 0 because I have cleaned them all, not sure how ip.ip_configuration returned 8. When I last check it, I see it was returning some false data.

Still I shall check once.

Reply:
I'm looking at other resource groups and subscriptions than you are. The important part is that the check should be against `ip.ip_configuration` to determine if its unused.

Reply:
Hmm! Not sure how I misguided first time and I decided to switch over to ip_address. The changes are addressed !